api/clients/eigenda_client.go

```diff
@@ -178,18 +199,34 @@ func (m *EigenDAClient) putBlob(ctx context.Context, rawData []byte, resultChan
 
 	alreadyWaitingForDispersal := false
 	alreadyWaitingForFinalization := false
+	var latestBlobStatus grpcdisperser.BlobStatus
 	for {
 		select {
 		case <-ctx.Done():
-			errChan <- fmt.Errorf("timed out waiting for EigenDA blob to confirm blob with request id=%s: %w", base64RequestID, ctx.Err())
+			// We can only land here if blob is still in
+			// 1. processing or dispersing status: waiting to land onchain
+			// 2. or confirmed status: landed onchain, waiting for finalization
+			// because all other statuses return immediately below.
+			//
+			// Assuming that the timeout is correctly set (long enough to both land onchain + finalize),
+			// 1. means that there is a problem with EigenDA, so we return 503 to let the batcher failover to ethda
+			// 2. means that there is a problem with Ethereum, so we return 500.
+			//    batcher would most likely resubmit another blob, which is not ideal but there isn't much to be done...
+			//    eigenDA v2 will have idempotency so one can just resubmit the same blob safely.
+			if latestBlobStatus == grpcdisperser.BlobStatus_PROCESSING || latestBlobStatus == grpcdisperser.BlobStatus_DISPERSING {
+				errChan <- api.NewErrorAPIGeneric(http.StatusServiceUnavailable,
+					fmt.Errorf("eigenda might be down. timed out waiting for blob to land onchain (request id=%s): %w", base64RequestID, ctx.Err()))
+			}
+			// but not else (otherwise it might be a problem with ethereum, so fallbacking to ethda wouldnt help)
```

Comment:
should we be more explicit, with else if (grpcdisperser.Confirmed)

and default to return something else. Maybe useful for debugging?

Reply:
good idea

Reply:
[5f8c3fb](https://github.com/Layr-Labs/eigenda/pull/828/commits/5f8c3fba836405ec937521367d36e5577bbf5d88)